test/clustertest/tests/MultipleLeaderSyncTest.cpp

```diff
@@ -0,0 +1,105 @@
+#include "../BedrockClusterTester.h"
+
+struct MultipleLeaderSyncTest : tpunit::TestFixture {
+    MultipleLeaderSyncTest()
+        : tpunit::TestFixture("MultipleLeaderSyncTest",
+                              BEFORE_CLASS(MultipleLeaderSyncTest::setup),
+                              AFTER_CLASS(MultipleLeaderSyncTest::teardown),
+                              TEST(MultipleLeaderSyncTest::test)
+                             ) { }
+
+    BedrockClusterTester* tester;
+
+    // Create a bunch of trivial write commands.
+    void runTrivialWrites(int writeCount, BedrockTester& node) {
+        int count = 0;
+        while (count <= writeCount) {
+            SData request;
+            request.methodLine = "Query";
+            if (count == 0) {
+                request["query"] = "INSERT OR REPLACE INTO test (id, value) VALUES(12345, 1 );";
+                node.executeWaitVerifyContent(request, "200");
+            } else {
+                request["query"] = "UPDATE test SET value=value + 1 WHERE id=12345;";
+                request["connection"] = "forget";
+                node.executeWaitVerifyContent(request, "202");
+            }
+            count++;
+        }
+    }
+
+    void setup() {
+        // create a 5 node cluster
+        tester = new BedrockClusterTester(ClusterSize::FIVE_NODE_CLUSTER, {"CREATE TABLE test (id INTEGER NOT NULL PRIMARY KEY, value TEXT NOT NULL)"});
+
+        // make sure the whole cluster is up
+        ASSERT_TRUE(tester->getTester(0).waitForStates({"LEADING", "MASTERING"}));
+        for (int i = 1; i <= 4; i++) {
+            ASSERT_TRUE(tester->getTester(i).waitForStates({"FOLLOWING", "SLAVING"}));
+        }
+    }
+
+    void teardown() {
+        delete tester;
+    }
+
+    void test() {
+        // get convenience handles for the cluster members
+        BedrockTester& node0 = tester->getTester(0);
```

Comment:
Does it make any difference to performance, creating a tester on-demand vs once at the top of the function?

Does it make sense to hide the behavior of the interface this way? Since you're not actually referring to an existing object at that position, won't this confuse engineers who don't know about the convenience function?